### DISCUSSION TITLE
Trying and failing to use whisper on window to generate subs

### DISCUSSION THREAD

**Discussion by BobHD** [openai/whisper#2461] [Q&A] [answered]
Hi,

I am new to this, so I used ChatGPT to guide me with the following steps:

```
pip install -U openai-whisper
choco install ffmpeg 
pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
Install: python-3.13.0-amd64.exe
```

If I run whisper:
`'C:\Users\m1\AppData\Local\Programs\Python\Python311\Scripts\whisper.exe audio.wav --model turbo'`

I get nice subs, but the time-stamps are wrong in a way impossible to correct them with SubtitleEdit unless aligning each of the subs one by one.

Chat told me to use this python script:

```python
import whisper
                                                                                                                 
# Load the Whisper model
model = whisper.load_model("base")  # Change "base" to "small", "medium", or "large" if needed
                                                                                                                 
# Transcribe the audio
result = model.transcribe("audio.wav", task="transcribe", word_timestamps=True)
                                                                                                                 
# Save the transcription as an SRT file
with open("output.srt", "w", encoding="utf-8") as f:
    f.write(result["srt"])
```

First it failed with:

```
C:\Users\m1\AppData\Local\Programs\Python\Python313\python.exe GenSubs.py
Traceback (most recent call last):
  File "A:\proa\m\GenSubs.py", line 1, in <module>
    import whisper
  File "c:\Users\m1\AppData\Local\Programs\Python\Python313\Lib\site-packages\whisper.py", line 70, in <
odule>
    libc = ctypes.CDLL(libc_name)
  File "c:\Users\m1\AppData\Local\Programs\Python\Python313\Lib\ctypes\__init__.py", line 380, in __init
_
    if '/' in name or '\\' in name:
       ^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
```
Chat me told to edit 'C:\Users\m1\AppData\Local\Programs\Python\Python313\Lib\site-packages\whisper.py' and replace `libc = ctypes.CDLL(libc_name)` with `libc = None`

And now I get:

`NT{M}% C:\Users\m1\AppData\Local\Programs\Python\Python313\python.exe GenSubs.py
Traceback (most recent call last):
  File "A:\proa\m\GenSubs.py", line 4, in <module>
    model = whisper.load_model("base")  # Change "base" to "small", "medium", or "large" if needed
            ^^^^^^^^^^^^^^^^^^
AttributeError: module 'whisper' has no attribute 'load_model'`

I'm stuck! 

Any help would be appreciated.

**Comment by glangford** [comment, marked as answer]
The simplest way to get an SRT with word timestamps is to use the command line. If you are using the latest whisper, try the turbo model first. For example,

`whisper audio.wav --model turbo --task transcribe --word_timestamps True`
`whisper audio1.wav audio2.mp3 video3.mp4 --model turbo --task transcribe --word_timestamps True`

You can see the full range of command line options with

`whisper --help`

I don't recommend chatGPT as your primary resource to understand whisper. Search these discussion boards first, there are a few years worth of questions and answers across different platforms.

**Reply by BobHD** [reply]
Thanks, Chat  is indeed limited, better to interact with humans here.

Something weird: the first caption is "Closed captioning by Ben Thede".
It's not in the audio. 
Where does that come from? Maybe a Whisper contributor?

With your settings, the time-stamps were almost perfect. They only needed a slight global adjustment with SubtileEdit.

I am pretty amazed by the result!

**Reply by glangford** [reply]
Glad it worked.

> Something weird: the first caption is "Closed captioning by Ben Thede".
It's not in the audio.

Yes, unfortunately you will see problems like this from time to time, it is discussed in detail here

* https://github.com/openai/whisper/discussions/928

**Reply by BobHD** [reply]
Interesting, that shows how an AI can do unexpected things.

**Reply by glangford** [reply]
There are two contributing factors - the AI can "hallucinate" on its own, but this whisper has also been trained on at least some subtitles that do not match the audio. So the model has learned to put things like closed captioning credits (with no matching audio) because the training dataset was not fully curated.
